src/tribler/gui/tribler_request_manager.py

```diff
@@ -188,6 +190,11 @@ def __init__(
         # Pass the newly created object to the manager singleton, so the object can be dispatched immediately
         request_manager.add_request(self)
 
+    def is_shutdown_request(self):
+        return "shutdown" in self.url \
+               and self.method == "PUT" \
+               and self.priority == QNetworkRequest.HighPriority
```

Comment:
NIT: Is it necessary to check a method and a priority to detect the shutdown request? I'm just wondering if the priority changes, then this request still is the shutdown request, right?

What if we derive the class `ShutdownRequest` from `TriblerNetworkRequest` like:
```python
class ShutdownRequest(TriblerNetworkRequest):
    def __init__(self, *args, **kwargs):
        super().__init__("shutdown", *args, **kwargs, method="PUT", priority=QNetworkRequest.HighPriority)
```

And use it in the cycle above:

```python
    def clear(self, skip_shutdown_request=True):
        for req in list(self.requests_in_flight.values()):
            if skip_shutdown_request and isinstance(req, ShutdownRequest):
                continue
            req.cancel_request()
```

In this case, the check is much more correct and future-proof.
The downside of this approach is that you should change all `shutdown` usage from

```python

                TriblerNetworkRequest("shutdown", shutdown_request_processed, method="PUT",
                                      priority=QNetworkRequest.HighPriority, on_cancel=send_shutdown_request)
```

to 

```python
                ShutdownRequest(shutdown_request_processed, on_cancel=send_shutdown_request)```

Reply:
It is a good idea, will make the changes.